perun-core/src/main/java/cz/metacentrum/perun/core/api/UsersManager.java

```diff
@@ -393,6 +393,18 @@
 	 */
 	List<UserExtSource> getUserExtSources(PerunSession perunSession, User user) throws InternalErrorException, UserNotExistsException, PrivilegeException;
 
+	/**
+	 * Gets list of all user's external sources with attributes.
+	 *
+	 * @param perunSession session
+	 * @param user user for which should be returned rich ext sources
+	 * @return list of user's external sources with attributes
+	 * @throws InternalErrorException internal error
+	 * @throws UserNotExistsException if given user doesn't exist
+	 * @throws PrivilegeException insufficient permissions
+	 */
+	List<RichUserExtSource> getRichUserExtSources(PerunSession perunSession, User user) throws InternalErrorException, UserNotExistsException, PrivilegeException;
```

Comment:
As we talked, please add also version, where you can specify list of attribute names to retrieve with each RichUserExtSource.